src/com/nisovin/magicspells/spells/targeted/MaterializeSpell.java

```diff
@@ -10,41 +14,126 @@
 import org.bukkit.block.BlockFace;
 import org.bukkit.block.BlockState;
 import org.bukkit.entity.Player;
+import org.bukkit.util.Vector;
 
 import com.nisovin.magicspells.DebugHandler;
 import com.nisovin.magicspells.MagicSpells;
 import com.nisovin.magicspells.events.MagicSpellsBlockPlaceEvent;
 import com.nisovin.magicspells.events.SpellTargetLocationEvent;
 import com.nisovin.magicspells.materials.MagicMaterial;
+import com.nisovin.magicspells.materials.MagicItemMaterial;
 import com.nisovin.magicspells.spelleffects.EffectPosition;
 import com.nisovin.magicspells.spells.TargetedLocationSpell;
 import com.nisovin.magicspells.spells.TargetedSpell;
 import com.nisovin.magicspells.util.compat.EventUtil;
+import com.nisovin.magicspells.util.BlockUtils;
 import com.nisovin.magicspells.util.HandHandler;
 import com.nisovin.magicspells.util.MagicConfig;
+import com.nisovin.magicspells.util.Util;
 
 public class MaterializeSpell extends TargetedSpell implements TargetedLocationSpell {
 
+	/*These extra features were inspired by Shadoward12's Rune/Pattern-Tester spell
+	Thank You! Shadoward12!*/
+
+	//Normal Features
 	MagicMaterial material;
+	Set<Material> materials;
 	private int resetDelay;
 	private boolean falling;
 	private boolean applyPhysics;
 	private boolean checkPlugins;
 	boolean playBreakEffect;
 	private String strFailed;
-	
+	private String blocktype;
+
+	//Pattern Configuration
+	private boolean usePattern;
+	private List<String> patterns;
+	private MagicMaterial[][] rowPatterns;
+	private boolean restartpatterneachrow;
+	private boolean randomizepattern;
+
+	//Cuboid Parameters
+	private String area;
+	private int height;
+	private double fallheight;
+
+	//Cubiod Variables;
+	private int rowSize;
+	private int columnSize;
+
+	//Cubiod Checks;
+	private boolean hasMiddle;
+
+	//Delays
+	private double blockdelay;
+
 	public MaterializeSpell(MagicConfig config, String spellName) {
 		super(config, spellName);
-		
-		String s = getConfigString("block-type", "1");
-		material = MagicSpells.getItemNameResolver().resolveBlock(s);
+
+		blocktype = getConfigString("block-type", "1");
```

Comment:
Why is this not using a method local variable? It doesn't seem to be used outside of the constructor.